[CI] HierarchicalKMeansTests testHKmeans failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9983 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/7w3fyuqy4jjko)
- [elasticsearch-periodic #9972 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/kkshlsmsfjyz6)
- [elasticsearch-pull-request #85354 / part-1](https://gradle-enterprise.elastic.co/s/r245uox3dna6w)
- [elasticsearch-pull-request #85299 / part-1](https://gradle-enterprise.elastic.co/s/eaia56du7rfhk)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.vectors.cluster.HierarchicalKMeansTests.testHKmeans" -Dtests.seed=F60947A932DC2C70 -Dtests.locale=ceb-PH -Dtests.timezone=EST -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.vectors.cluster.HierarchicalKMeansTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testHKmeans'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 4 failures in test testHKmeans (1.8% fail rate in 225 executions)
- [main] 2 failures in step part-1 (1.8% fail rate in 111 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.8% fail rate in 111 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -504,9 +504,6 @@ tests:
 - class: org.elasticsearch.repositories.SnapshotMetricsIT
   method: testSnapshotAPMMetrics
   issue: https://github.com/elastic/elasticsearch/issues/132731
-- class: org.elasticsearch.index.codec.vectors.cluster.HierarchicalKMeansTests
-  method: testHKmeans
-  issue: https://github.com/elastic/elasticsearch/issues/132771
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeForkIT
   method: test {csv-spec:lookup-join.MvJoinKeyFromRowExpanded}
   issue: https://github.com/elastic/elasticsearch/issues/132778

--- a/server/src/test/java/org/elasticsearch/index/codec/vectors/cluster/HierarchicalKMeansTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/vectors/cluster/HierarchicalKMeansTests.java
@@ -46,8 +46,9 @@ public class HierarchicalKMeansTests extends ESTestCase {
             assertEquals(nVectors, soarAssignments.length);
             // verify no duplicates exist
             for (int i = 0; i < assignments.length; i++) {
-                assertTrue(soarAssignments[i] >= 0 && soarAssignments[i] < centroids.length);
-                assertNotEquals(assignments[i], soarAssignments[i]);
+                int soarAssignment = soarAssignments[i];
+                assertTrue(soarAssignment == -1 || (soarAssignment >= 0 && soarAssignment < centroids.length));
+                assertNotEquals(assignments[i], soarAssignment);
             }
         } else {
             assertEquals(0, soarAssignments.length);

